services/keyword_service/keyword_test.go

```diff
@@ -86,6 +86,47 @@ func (s *KeywordServiceDbTestSuite) TestGetAllWithInvalidUser() {
 	assert.Equal(s.T(), nil, err)
 }
 
+func (s *KeywordServiceDbTestSuite) TestGetKeywordResultWithValidKeywordAndUser() {
+	keyword := models.Keyword{UserID: s.userID, Keyword: faker.Name()}
+	db.GetDB().Create(&keyword)
+
+	result, err := s.keywordService.GetKeywordResult(keyword.ID)
+
+	assert.Equal(s.T(), keyword.Keyword, result.Keyword)
+	assert.Equal(s.T(), nil, err)
+}
+
+func (s *KeywordServiceDbTestSuite) TestGetKeywordResultWithValidKeywordButInvalidUser() {
+	keyword := models.Keyword{UserID: s.userID, Keyword: faker.Name()}
+	db.GetDB().Create(&keyword)
+
+	keywordService := KeywordService{CurrentUserID: 999999}
+	result, err := keywordService.GetKeywordResult(keyword.ID)
+
+	assert.Equal(s.T(), models.Keyword{}, result)
+	assert.Equal(s.T(), "something went wrong, please try again", err.Error())
+}
+
+func (s *KeywordServiceDbTestSuite) TestGetKeywordResultWithWrongKeywordTypeButValidUser() {
+	keyword := models.Keyword{UserID: s.userID, Keyword: faker.Name()}
+	db.GetDB().Create(&keyword)
+
+	result, err := s.keywordService.GetKeywordResult("invalidKeyword")
+
+	assert.Equal(s.T(), models.Keyword{}, result)
+	assert.Equal(s.T(), "ERROR: invalid input syntax for type bigint: \"invalidKeyword\" (SQLSTATE 22P02)", err.Error())
```

Comment:
This kind of error messages are really not user friendly 🙈 No way with GORM to have a better message like "keyword not found for ..." Or handle it at the service level.

Saem above for `TestGetKeywordResultWithValidKeywordButInvalidUser` as it should just return a similar error

Reply:
Fixed in c08a6c9c7ba927707ba5830bf53c8a1078368a55